bigchaindb/backend/rethinkdb/config.py

```diff
@@ -0,0 +1,48 @@
+"""Database configuration functions."""
+import logging
+from functools import partial
+
+import rethinkdb as r
+
+from bigchaindb.backend import config
+from bigchaindb.backend.exceptions import DatabaseOpFailedError
+from bigchaindb.backend.rethinkdb.connection import RethinkDBConnection
+
+logger = logging.getLogger(__name__)
+
+TABLES = ('bigchain', 'backlog', 'votes')
+
+
+def _configure_table_helper(connection, table, *, replicas=None, shards=None):
+    try:
+        connection.run(
+            r.table(table).reconfigure(shards=shards, replicas=replicas)
+        )
+    except r.ReqlOpFailedError as e:
+        raise DatabaseOpFailedError from e
+
+
+def _configure_table(connection, table, *, replicas=None, shards=None):
+    if replicas is None:
+        table_config = connection.run(r.table(table).config())
+        replicas = len(table_config['shards'][0]['replicas'])
+    if shards is None:
+        table_config = connection.run(r.table(table).config())
+        shards = len(table_config['shards'])
+    _configure_table_helper(
+        connection, table, replicas=replicas, shards=shards)
+
+_configure_table_shards = partial(_configure_table, replicas=None)
+_configure_table_replicas = partial(_configure_table, shards=None)
```

Comment:
Well, two things;

* The code for each, `set_shards` and `set_replicas` is nearly identical, and very repetitive. So this was an attempt to limit the duplication of identical logic. 

* From the point of view of an API, it appears that `set_shards` and `set_replicas` are just specialized commands, or even special cases of the more general task of configuring tables. Even this is limited, and could be further generalized as being admin tasks. (From the point of view of rethinkdb at least, `reconfigure` is part of the `Administration` group, which includes other commands (`grant`, `config`, `rebalance`, `status`, `wait`).

So from the point of view of the two above points, I do not think that the proposed code is point-less, it is at the very least point-two. :smile: